### PR TITLE
Use cffi 1.X features to build native module for faster import

### PR DESCRIPTION
**Pull request by devinlane**
#41 This uses the cffi 1.X set_source feature to build a native module, which can then be quickly imported. It falls back to the legacy ffi.verify() if the module could not be loaded. Tested on Linux  x86_64.

### CODE REVIEW

**Comment by devinlane** [comment]
Builds seem to be failing with errors like "python-build: definition not found: 2.7.11"

**Comment by alex** [comment]
@reaperhulk can I bug you to help review this?

**Comment by devinlane** [comment]
Please note that there is no change to the actual xattr C code -- it is just moved to another file.

File: xattr/lib_build.py [hunk 1]
```diff
@@ -0,0 +1,602 @@
...
+#endif
+
+#ifndef XATTR_MAXNAMELEN
+#define XATTR_MAXNAMELEN 127
+#endif
+"""
+
+ffi.set_source('_lib', C_SRC)
+
+if __name__ == '__main__':
```
Comment (reaperhulk) [inline xattr/lib_build.py]: Unless you intend to invoke this file manually you don't really need this conditional. Generally you just need the `ffi` variable out of this module, which cffi/setuptools does in the setup.py.

**Comment by reaperhulk** [comment]
(Reviewing at @alex's request). This looks mostly fine to me. The issue with Travis is that you need to `brew update && brew upgrade pyenv` because the version of pyenv shipped doesn't have the Python versions you need. Additionally, cffi 1.0 is not supported on pypy3 (it ships with 0.8.6) so you won't be able to test it against pypy3 (or pypy < 2.6).

File: xattr/lib_build.py (hunk 1 above)
Comment (devinlane) [inline xattr/lib_build.py]: By keeping it in there, you can do the lib generation manually which is helpful for debugging. Seems easier than having to run the build script every time. Of course I can remove if you disagree.

Comment (reaperhulk) [inline xattr/lib_build.py]: No strong opinion at all, if you find it useful keep it in :)

**Comment by devinlane** [comment]
@reaperhulk I'm not even sure what pypy is, do you want to maintain compatibility with it? It sounds like merging this patch would break use with pypy3 or pypy < 2.6. I'm assuming that the homebrew stuff is something you guys need to do?

**Comment by reaperhulk** [comment]
@devinlane pypy is just an alternative Python implementation (as opposed to CPython, the most common one). @alex do you have an opinion on how CI should be fixed? It seems like it probably needs to be fixed up in a separate PR and then this can be retested?

**Comment by devinlane** [comment]
Have there been any updates on this?

**Comment by devinlane** [comment]
Ping :)

**Comment by devinlane** [comment]
Ping 2 :)

**Comment by alex** [comment]
I skimmed this, and it looks ok, but the tests aren't passing on travis

**Comment by devinlane** [comment]
@alex The test failure doesn't seem to have anything to do with this pull request. The error indicates installing a specific version of python using pyenv failed. @reaperhulk mentioned a solution to the problem above, but having not used travis I'm not sure how to fix it.

**Pull request merged**